pkg/goDB/engine/aggregate.go

```diff
@@ -58,18 +70,34 @@ func (qr *QueryRunner) aggregate(mapChan <-chan hashmap.AggFlowMapWithMetadata,
 			finalMaps = hashmap.NewNamedAggFlowMapWithMetadata(ifaces)
 		)
 
+		tLastStatsUpdate := time.Now()
 		for item := range mapChan {
 			if item.IsNil() || item.Interface == "" {
 				resultChan <- aggregateResult{err: errorInternalProcessing}
 				return
 			}
 
 			finalMap := finalMaps[item.Interface]
+			finalMap.Stats.Add(item.Stats)
+
+			// the processing stats have been processed. Skip to next item in case there's no flow data to process. This
+			// is relevant for cases where no flow records are returned as a result of conditions not matching
+			if item.Len() == 0 {
+				continue
+			}
 
 			// Merge the item into the final map for this interface
 			finalMap.Merge(item)
 			nAgg[item.Interface] = nAgg[item.Interface] + 1
 
+			// keep-alive updating of queries
+			if qr.keepAlive > 0 {
```

Comment:
This is the actual keep-alive mechanism: now only bound by time, not by interface.